test-suite/shared/src/test/require-jdk8/org/scalajs/testsuite/javalib/time/YearMonthTest.scala

```diff
@@ -0,0 +1,442 @@
+    assertEquals(12L, janLastBC.until(janFirstAC, MONTHS))
+    assertEquals(1L, janLastBC.until(decFirstAC, YEARS))
+    assertEquals(23L, janLastBC.until(decFirstAC, MONTHS))
+    assertEquals(0L, decLastBC.until(janFirstAC, YEARS))
+    assertEquals(1L, decLastBC.until(janFirstAC, MONTHS))
+
+    for {
+      ym1 <- samples
+      ym2 <- samples if ym2.isAfter(ym1)
+      unit <- ChronoUnit.values() if isSupported(unit)
+    } {
+      assertEquals(-ym1.until(ym2, unit), ym2.until(ym1, unit))
+    }
+  }
+
+  @Test def atDay(): Unit = {
+    for {
+      ym <- samples
+      day <- 1 to ym.lengthOfMonth()
+    } {
+      assertEquals(LocalDate.of(ym.getYear, ym.getMonthValue, day), ym.atDay(day))
+    }
+
+    for (ym <- samples) {
+      expectThrows(classOf[DateTimeException], ym.atDay(0))
+      expectThrows(classOf[DateTimeException], ym.atDay(ym.lengthOfMonth() + 1))
+    }
+  }
+
+  @Test def atEndOfMonth(): Unit = {
+    for (ym <- samples) {
+      val endOfMonth = ym.lengthOfMonth()
+      assertEquals(LocalDate.of(ym.getYear, ym.getMonthValue, endOfMonth), ym.atEndOfMonth())
+    }
+  }
+
+  @Test def compareTo(): Unit = {
+    assertEquals(0, min.compareTo(min))
+    assertEquals(-1999999998, min.compareTo(max))
+    assertEquals(-11, min.compareTo(YearMonth.of(Year.MIN_VALUE, 12)))
+    assertEquals(1999999998, max.compareTo(min))
+    assertEquals(11, max.compareTo(YearMonth.of(Year.MAX_VALUE, 1)))
+    assertEquals(0, max.compareTo(max))
+  }
+
+  @Test def isAfter(): Unit = {
+    assertFalse(min.isAfter(min))
+    assertFalse(min.isAfter(max))
+    assertTrue(max.isAfter(min))
+    assertFalse(max.isAfter(max))
+  }
+
+  @Test def isBefore(): Unit = {
+    assertFalse(min.isBefore(min))
+    assertTrue(min.isBefore(max))
+    assertFalse(max.isBefore(min))
+    assertFalse(max.isBefore(max))
+  }
+
+  @Test def toStringOutput(): Unit = {
+    assertEquals("-999999999-01", min.toString)
+    assertEquals("999999999-12", max.toString)
+    assertEquals("-10000-01", YearMonth.of(-10000, 1).toString)
+    assertEquals("10000-12", YearMonth.of(10000, 12).toString)
+    assertEquals("2015-06", someYearMonth.toString)
+    assertEquals("0000-01", janLastBC.toString)
+    assertEquals("0000-12", decLastBC.toString)
+    assertEquals("0001-01", janFirstAC.toString)
+    assertEquals("0001-12", decFirstAC.toString)
+  }
+
+  @Test def ofMonth(): Unit = {
+    val yearMonth = YearMonth.of(23, Month.JANUARY)
+    assertEquals(23, yearMonth.getYear)
+    assertEquals(1, yearMonth.getMonthValue)
+    assertEquals(Month.JANUARY, yearMonth.getMonth)
+
+    for (ym <- samples) {
+      assertEquals(ym, YearMonth.of(ym.getYear, ym.getMonth))
+    }
+
+    expectThrows(classOf[NullPointerException], YearMonth.of(0, null))
+    for (m <- Month.values()) {
+      expectThrows(classOf[DateTimeException], YearMonth.of(Int.MinValue, m))
+      expectThrows(classOf[DateTimeException], YearMonth.of(Int.MaxValue, m))
+    }
+  }
+
```

Comment:
Aren't all the different assertions across the test, plus that `assertEquals` in the `of` tests, already testing the `equals` method?